### PR TITLE
Add first fuzz target for fuzzing simplify algorithm.

### PR DESCRIPTION
**Pull request by frewsxcv**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- ~I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.~
---

Based off of https://github.com/georust/geo/pull/532.

Fuzzing allows us to find buggy edge cases and  alsocheck the validity of our algorithms (by comparing results against the result of other libraries like GEOS). This pull request just adds one fuzz target for fuzzing our `Simplify` algorithm to get us started.

How to fuzz:

```sh
cargo install cargo-fuzz
cd geo
cargo fuzz run simplify # cargo fuzz run <fuzz target>
```

### CODE REVIEW

File: geo/fuzz/fuzz_targets/simplify.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#![no_main]
+
+use geo::algorithm::simplify::Simplify;
+use libfuzzer_sys::fuzz_target;
+
+fuzz_target!(|tuple: (geo_types::Polygon<f32>, f32)| {
+    if tuple.1 < 0. {
+        return;
+    }
+    tuple.0.simplify(&tuple.1);
```
Comment (michaelkirk) [inline geo/fuzz/fuzz_targets/simplify.rs:10]: So basically, we're only testing that no `panic` occurs. Is that right? That seems helpful.

Do you think it's worth checking additional invariants?

I couldn't come up with much... maybe:
- number of coords in the output <= input's
- the euclidean length of the output's linestrings should be <= input's

I'd also be happy to leave it as is.

**Review by michaelkirk — APPROVED** [review]
LGTM, just some "take it or leave it" feedback.

File: geo/fuzz/fuzz_targets/simplify.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#![no_main]
+
+use geo::algorithm::simplify::Simplify;
+use libfuzzer_sys::fuzz_target;
+
+fuzz_target!(|tuple: (geo_types::Polygon<f32>, f32)| {
+    if tuple.1 < 0. {
```
[the remark this answers is not in the document]
Comment (frewsxcv) [reply geo/fuzz/fuzz_targets/simplify.rs]: https://github.com/georust/geo/pull/633/commits/1a1942e0721d27033734bd1c87d73f93b0152f00

File: geo/fuzz/fuzz_targets/simplify.rs (hunk 1 above)
Comment (frewsxcv) [reply geo/fuzz/fuzz_targets/simplify.rs]: https://github.com/georust/geo/pull/633/commits/dd4139d2d841dad4378725ae599c242332570d66

**Comment by frewsxcv** [comment]
bors r=michaelkirk

**Pull request merged**